src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
@stephentoub - looks like the answer is "probably not measurable" - there's some slight differences in timing, but sometimes the checked version is faster, and even completely removing the checking can be slower (for whatever reason).

I used the code [in this branch](https://github.com/Clockwork-Muse/corefx/tree/PLINQ_Perf), creating another test.
[Current PLINQ results](https://onedrive.live.com/redir?resid=9AD89593899AA409!38191&authkey=!AJv9-cuHQBnbsjA&ithint=file%2cxlsx)
[Full Checking results](https://onedrive.live.com/redir?resid=9AD89593899AA409!38192&authkey=!AGn2D1U0yhu9w6w&ithint=file%2cxlsx)
[Long-addition-only-checking results](https://onedrive.live.com/redir?resid=9AD89593899AA409!38193&authkey=!AKmO0e60BBEw6EQ&ithint=file%2cxlsx) (for a lark).

(I'm aware this might not have been the best way to do this, but it was quick to figure out, so...)